### PR TITLE
[IMP] product_barcodelookup: product template view improvements

### PR DESCRIPTION
**Pull request by sjai-odoo**
[IMP] account,product,website_sale: add helpers for barcodelookup

This commit adds a hooks to compute a gross price based on product
taxes. This is needed for `product_barcodelookup` module.
Also change a wording in the ecommerce product template

Related: https://github.com/odoo/enterprise/pull/63740
Task: 3965079

### CODE REVIEW

**Review by chevalierv — APPROVED** [review]

File: addons/product/models/product_template.py [hunk 1]
```diff
@@ -1463,3 +1463,7 @@ def _get_contextual_pricelist(self):
         This method is meant to be overriden in other standard modules.
         """
         return self.env['product.pricelist'].browse(self.env.context.get('pricelist'))
+
+    def _get_gross_list_price(self, price):
```
Comment (malb-odoo) [inline addons/product/models/product_template.py:1467]: Suggestion: From what i'm seeing, the function will be override by other module so could be nice to have a docstring to explain in which condition we are using it and why we may need to override this ? :smile:

File: addons/account/models/product.py [hunk 2]
```diff
@@ -139,6 +139,19 @@ def create(self, vals_list):
         products.invalidate_recordset(['taxes_id', 'supplier_taxes_id'])
         return products
 
+    def _get_gross_list_price(self, price):
```
Comment (malb-odoo) [inline addons/account/models/product.py:142]: also here a docstring would be nice to explain the function

File: addons/account/models/product.py [hunk 3]
```diff
@@ -139,6 +139,19 @@ def create(self, vals_list):
...
+        self.ensure_one()
+        if self.taxes_id:
```
Comment (malb-odoo) [inline addons/account/models/product.py:144]: if we do this: 
```
if not self.taxes_id:
    return price
```

we can gain one level of indentation so that the function is more readable

**Review by malb-odoo — COMMENTED** [review]
thanks for this PR ! :smile: 
Some suggestions to make the code a bit better, also i found the commit message a bit light to explain the change you have done. 
In the pr description: 
You only mention ecommerce description in product template but you are also adding a method to get a price in product and accounting ? 
Also the renaming in website sale not mention in the commit message ?

In the commit message: only the add of the hooks but nothing about the description and the renaming

Also PR title could be modified aswell to match the commit title :smile:

File: addons/account/models/product.py (hunk 2 above)
Comment (Feyensv) [inline addons/account/models/product.py:142]: If those methods are only used in barcodelookup module, I think they could be defined directly in that module and not here :thinking:

File: addons/website_sale_comparison/views/website_sale_comparison_template.xml [hunk 4]
```diff
@@ -44,7 +44,7 @@
                                                 <tr>
                                                     <th class="text-start" t-att-colspan="2">
                                                         <span t-if="category" t-field="category.name"/>
-                                                        <span t-else="">Uncategorized</span>
+                                                        <span t-else="">Others</span>
```
Comment (Feyensv) [inline addons/website_sale_comparison/views/website_sale_comparison_template.xml:47]: How is this change related to the barcodelookup task ? :thinking:

Comment (Whenrow) [reply addons/website_sale_comparison/views/website_sale_comparison_template.xml:47]: It was asked in the [Excalidraw](https://app.excalidraw.com/l/65VNwvy7c4X/3Z4WpFU8tMN)
image

File: addons/account/models/product.py (hunk 2 above)
Comment (Whenrow) [reply addons/account/models/product.py:142]: `barcodelookup` don't depend on `account` but needs to compute the price based on taxes. Do you prefer a bridge module instead ?

Comment (Feyensv) [reply addons/account/models/product.py:142]: Could have made sense, but not worth postponing something required for the OCD, leave it here then (and thanks for the explanation).

**Comment by Whenrow** [comment]
robodoo r+
